File: 8th/04.19.2019_Упорядочные_деревья_2/pr2.ml
open String;;
type tree = N of int * tree * tree|L;;


let rec make_whl n t = 
if n = 0 then L else N (t,N (t-1,L,L),make_whl (n-1) (t + 2));;

let rec tr_to_str_lkp tr = 
match tr with
 L -> ""
|N (k,tr1,tr2) -> "("^(tr_to_str_lkp tr1)^","^(string_of_int k)^","^(tr_to_str_lkp tr2)^")";;


print_string (tr_to_str_lkp (make_whl (read_int()) 2));; 

File: 9th/PYHTHON/20.05.2020 HW/pr2.py
import re

names = {}

f =  open("names","r")
for x in f:
	g = re.match(r'^(.+)@(.+)\.(.+)',x)
	if g:
		names[g.group(2)] = names.setdefault (g.group(2),[]) + [g.group(1)]

print (names)